### PR TITLE
Fix memory leak in Context class

### PR DESCRIPTION
**Pull request by tszumski**
Fix memory leak in Context class

### CODE REVIEW

File: media-proxy/src/mesh/concurrency.cc [hunk 1]
```diff
@@ -55,9 +55,11 @@ Context& Context::operator=(Context&& other) noexcept {
         ss = std::stop_source();
         parent = other.parent;
 
-        cb = std::make_unique<std::stop_callback<std::function<void()>>>(
-            parent->ss.get_token(), [this] { cancel(); }
-        );
+        if(parent){
```
[the remark this answers is not in the document]
Comment (tszumski) [reply media-proxy/src/mesh/concurrency.cc]: Fixed in https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/269/commits/a4cb2a96c2db986e12fc249f66e0654509fcb401

File: media-proxy/src/mesh/concurrency.cc [hunk 2]
```diff
@@ -14,7 +14,7 @@ namespace context {
 Context::Context() : ss(std::stop_source()),
                      parent(nullptr),
                      cb(nullptr),
-                     ch(new thread::Channel<bool>(1)),
+                     ch(nullptr),
```
[the remark this answers is not in the document]
Comment (tszumski) [reply media-proxy/src/mesh/concurrency.cc]: Fixed in https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/269/commits/a4cb2a96c2db986e12fc249f66e0654509fcb401

File: media-proxy/tests/mesh_context_tests.cc [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+};
+
+/**
+ * Test designed to detect memory leaks in the Context class.
+ *
+ * Run the test with Valgrind to detect memory leaks.
+ * valgrind --leak-check=full ./media_proxy_unit_tests --gtest_filter=*Context*
+ */
+TEST(MeshContext, constructor) {
+    auto ctx = context::WithCancel(mesh::context::Background());
```
Comment (ko80) [inline media-proxy/tests/mesh_context_tests.cc:20]:
```suggestion
    auto ctx = context::WithCancel(context::Background());
```

File: media-proxy/tests/mesh_context_tests.cc [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    test_class t;
+    t.init(ctx);
+    {
+        auto ctx2 = context::WithCancel(ctx);
+        t.init(ctx2);
+        mesh::thread::Sleep(ctx2, std::chrono::milliseconds(10));
```
Comment (ko80) [inline media-proxy/tests/mesh_context_tests.cc:26]: I think this test case was supposed to verify the `context` implementation only. Maybe it's worth replacing `thread::Sleep` with `std::chrono::Sleep`?

Comment (ko80) [reply media-proxy/tests/mesh_context_tests.cc:26]: But in general, I'm okay with using `thread::Sleep`.

Comment (tszumski) [reply media-proxy/tests/mesh_context_tests.cc:26]: Lets keep it as is

File: media-proxy/tests/mesh_context_tests.cc (hunk 3 above)
Comment (tszumski) [reply media-proxy/tests/mesh_context_tests.cc]: Fixed in https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/269/commits/c89d2973e0e8b88369250f85b1524cb66d0ae875

**Review by ko80 — APPROVED** [review]
Looks good. `mesh::` is redundant.

**Pull request merged**